pageserver/src/tenant/storage_layer/layer.rs

```diff
@@ -1050,15 +1050,15 @@ impl LayerInner {
                     LAYER_IMPL_METRICS.inc_eviction_cancelled(EvictionCancelled::LayerGone);
                     return;
                 };
-                match this.evict_blocking(version) {
+                match tokio::runtime::Handle::current().block_on(this.evict_blocking(version)) {
```

Comment:
this isn't great obviously. I wonder if we can add `get_mut_blocking` copy of `get_mut` which calls the `blocking_*` function on the `RwLock` instead.

Reply:
The `blocking_*` methods would do exactly the same.